packages/flutter/lib/src/material/slider.dart

```diff
@@ -505,12 +516,10 @@ class _RenderSlider extends RenderBox {
 
   void _updateLabelPainter() {
     if (label != null) {
-      // We have to account for the text scale factor in the supplied theme.
-      final TextStyle style = _theme.accentTextTheme.body2
-          .copyWith(fontSize: _theme.accentTextTheme.body2.fontSize * _textScaleFactor);
       _labelPainter
-        ..text = new TextSpan(style: style, text: label)
+        ..text = new TextSpan(style: _theme.accentTextTheme.body2, text: label)
         ..textDirection = textDirection
+        ..textScaleFactor = _mediaQueryData?.textScaleFactor ?? 1.0
```

Comment:
can you require the media query, rather than having to worry about fallback?

Reply:
As in, use `assert(debugCheckHasMediaQuery(context));` in the build method (and document appropriately). See, e.g., SafeArea.

Reply:
I could, and I had it that way, but we don't currently require it, so it may break some people if they're using a slider outside of a media query (and it currently breaks all the tests for slider, but I can fix that, of course). If you're fine with breaking people that do that, then I'm happy to do it that way.

Reply:
I spoke to Ian, and he said it's fine to break people who use a Material slider outside of a MaterialApp or other context where a MediaQuery is provided.

Reply:
you can remove the `?` and `?? 1.0`